python/preemo/worker/_worker_client.py

```diff
@@ -41,16 +29,17 @@ def decorator(function: Callable) -> Callable:
                 function, name=function_name, namespace=namespace
             )
 
-            worker_reply = self._client.send_worker_request(
-                _construct_register_function_worker_request(
-                    name=function_name, namespace=namespace
+            response = self._client.register_function(
+                RegisterFunctionRequest(
+                    function_to_register=RegisteredFunction(
+                        name=function_name, namespace=namespace
+                    )
                 )
             )
 
-            reply = worker_reply.register_function
-            if reply.status != STATUS_OK:
+            if response.status != STATUS_OK:
                 raise Exception(
-                    f"worker server replied to register function request with unexpected status: {reply.status} and message: {reply.message}"
+                    f"worker server replied to register function request with unexpected status: {response.status} and message: {response.message}"
```

Comment:
i feel like you usually you would write a switch here
```python 
match status:
 case STATUS_OK:
   yay
 case SOMETHING | ELSE:
   raise ...
 default:
   assert_never(status)
```

Reply:
yeah i was considering that. It's probably unnecessary here since it's likely we will only ever care about anything other than status ok, but i'll make the change regardless

Reply:
since assert_never doesn't work well with these fake enums, i'm going to leave it as is for now